tests/test_events.py

```diff
@@ -181,12 +180,87 @@ def test_put(self):
 
-    @skip("Unimplemented test")
+    def test_put_auth(self):
+        response = self.app.get('/events/?start=2017-01-01&end=2017-07-01')
+        self.assertEqual(response.status_code, 200)
+        event_id = flask.json.loads(response.data)[0]['id']
+
+        with self.subTest("fails when the client is not yet authorized"):
+            response = self.app.put(
+                f'/events/{event_id}',
+                data=flask.json.dumps({'title': 'new title'}),
+                content_type='application/json',
+                headers={
+                    'X-Forwarded-For': '192.168.1.1',
+                }
+            )
+            self.assertEqual(response.status_code, 401)
+
+        with self.subTest("succeeds when required fields are present"):
+            response = self.app.put(
+                f'/events/{event_id}',
+                data=flask.json.dumps({'title': 'new title'}),
+                content_type='application/json'
+            )
+            self.assertEqual(response.status_code, 200)
+            self.assertIn(b'new title', response.data)
+
+        with self.subTest("succeeds due to auth cookie"):
+            response = self.app.put(
+                f'/events/{event_id}',
+                data=flask.json.dumps({'title': 'new title'}),
+                content_type='application/json',
+                headers={
+                    'X-Forwarded-For': '192.168.1.1',
+                }
+            )
+            self.assertEqual(response.status_code, 200)
+
     def test_delete(self):
-        # TODO: test success
-        # TODO: test invalid id
         # TODO: test invalid data
-        # TODO: test unauthorized user
-        pass
+        response = self.app.get('/events/?start=2017-01-01&end=2017-07-01')
+        self.assertEqual(response.status_code, 200)
+        event_id = flask.json.loads(response.data)[0]['id']
+
+        with self.subTest("succeeds on valid id"):
+            response = self.app.delete(
+                f'/events/{event_id}'
+            )
+            self.assertEqual(response.status_code, 200)
+
+        with self.subTest("fails on invalid id"):
+            response = self.app.delete(
+                f'/events/{event_id}x'
+            )
+            # FIXME: why is this not 404?
+            self.assertEqual(response.status_code, 400)
+
+    def test_delete_auth(self):
+        response = self.app.get('/events/?start=2017-01-01&end=2017-07-01')
+        self.assertEqual(response.status_code, 200)
+        event_id = flask.json.loads(response.data)[0]['id']
+
+        with self.subTest("fails when the client is not yet authorized"):
+            response = self.app.delete(
+                f'/events/{event_id}',
+                headers={
+                    'X-Forwarded-For': '192.168.1.1',
+                }
+            )
+            self.assertEqual(response.status_code, 401)
+
+        with self.subTest("succeeds when required fields are present"):
```

Comment:
This subtest description is misleading. It actually tests whether a user inside the local network can delete an event.